### PR TITLE
Fix: Handle unnamed index/dimension in wave.resource.surface_elevation 

### PR DESCRIPTION
**Pull request by simmsa**
Fix: If unnamed, set spectrum dim_0 name to "Frequency"

The surface_elevation function typically takes in a spectrum generated
by an mhkit function that sets the index/dimension name to "Frequency".
This fix handles an atypical case where the index/first dimension is
unnamed, which causes downstream issues where we are expecting a
dimension named "Frequency".

Per the [xarray DataArray documentation](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.html)
an unnamed dimension starts at dim_0. If we find this case we
change the dim_0 name to "Frequency".

### CODE REVIEW

**Comment by akeeste** [comment]
@simmsa Can you describe further the case that causes this issue? I can see how this will be a problem, but ``surface_elevation`` does not return any info with a frequency dimension, so renaming it here will not fix the downstream workflow.

Is there another ``wave.resource`` function that creates but does not properly name the frequency dimension? In #336 I propagated the variable ``frequency_dimension`` further so that we always follow the user input.

**Review by akeeste — APPROVED** [review]
Thanks @simmsa for catching this. This PR looks good to me. Were there any more changes to add before this is merged?

**Comment by akeeste** [comment]
@simmsa Sorry about the delay here. Based on one of our conversations I wasn't sure if this was 100% done. It looks good as is. I'll merge this into the develop branch after v0.8.2 is released.

**Comment by simmsa** [comment]
@akeeste, no worries, it looks like I didn't follow up with you here.

This fixes a potential bug where a user inputs `S` without frequency index named "Frequency" by using the found or defined `frequency_dimension` variable: https://github.com/MHKiT-Software/MHKiT-Python/blob/406599e3129c1090d522dc806a4ac0493a360f71/mhkit/wave/resource.py#L303-L304 to define names of the DataArray dimension and coordinates. 

Without this fix, passing a DataFrame with an unnamed index yields: 

`ValueError: coordinate Frequency has dimensions ('dim_0',), but these are not a subset of the DataArray dimensions ['Time', 'Frequency']`

The tests verify that both computation methods, `ifft` and `sum_of_sines` work as expected, and verify that a user defined dataframe also works as expected.

Happy to clarify further if necessary.

**Pull request merged**
